sklearn/neighbors/tests/test_nca.py

```diff
@@ -531,18 +531,19 @@ def test_parameters_valid_types(param, value):
     nca.fit(X, y)
 
 
-def test_nca_feature_names_out():
+@pytest.mark.parametrize("n_components", range(1, iris.data.shape[1]))
```

Comment:
Only 2 values would be enough here: the default and when setting `n_components`.

Reply:
I added a test for `n_components` when it is `None`. Not sure if that was what you intended or I misunderstood.